spec/voting/voting.spec.ts

```diff
@@ -2,64 +2,240 @@ import * as ganache from 'ganache-cli'
+  beforeAll(async () => {
+    accounts = await w3.eth.getAccounts()
 
-  })
+    // deploy it...
+    const bin:string = readBytecode('voting')
 
-  afterAll(() => {
+    deployed = await deploy(w3,
+                            accounts[0],
+                            VOTING_ABI,
+                            bin,
+                            [accounts[0]])
 
+    // now we can instantiate the HOC
+    voting = new Voting(accounts[0])
+    await voting.at(w3, deployed.options.address)
   })
 
-  beforeEach(async () => {
-    accounts = await w3.eth.getAccounts()
+  describe('Class methods for Voting', () => {
 
-    // erc20 = new Erc20(accounts[0])
-    // voting = new Voting(accounts[0])
-    // parameterizer = new Parameterizer(accounts[0])
-    // listing = new Listing(accounts[0])
-
-    // 0th account approves voting and reg to spend
-    // await erc20.approve(web3, votingAddress, 1000000)
-    // await erc20.approve(web3, registryAddress, 1000000)
-    // 1st account, as challenger, needs funds
-    // await erc20.transfer(web3, accounts[1], 500000)
-    // 2nd account as voter needs funds
-    // await erc20.transfer(web3, accounts[2], 500000)
-    // registry needs to be approved to spend ond the challenger's behalf
-    // await erc20.approve(web3, registryAddress, 450000, { from: accounts[1] })
-    // voting needs approval from voter
-    // await erc20.approve(web3, votingAddress, 450000, { from: accounts[2] })
-  })
+    it('calls setPrivileged correctly', async () => {
+      const defaults = await voting.setPrivileged("listing", "reserve", {})
+      let tx = defaults[0]
+      let opts = defaults[1]
+      expect(tx).not.toBeNull(accounts[0])
+      expect(opts).not.toBeNull()
+      expect(Object.keys(opts).indexOf('gas')).toBeGreaterThan(-1)
+      expect(Object.keys(opts).indexOf('from')).toBeGreaterThan(-1)
+      expect(Object.keys(opts).indexOf('gasPrice')).toBeGreaterThan(-1)
+      expect(opts['gas']).toBeGreaterThanOrEqual(101191)
+      let from = opts['from']
+      expect(from).not.toBeNull()
+      expect(from!.trim().length).toBeGreaterThan(0)
+      let gasPrice = opts['gasPrice']
+      expect(gasPrice).not.toBeNull()
+      expect(parseInt(gasPrice)).toBeGreaterThan(0)
+    })
 
-  it('can be instantiated from an existing deployment', async () => {
-    // const address = voting.getAddress(),
-      // other = new Voting(accounts[0]),
-      // works = await other.at(web3, { address })
+    it('calls getPrivileged correctly', async () => {
+      const defaults = await voting.getPrivileged({})
+      let tx = defaults[0]
+      let opts = defaults[1]
+      expect(tx).not.toBeNull(accounts[0])
+      expect(opts).not.toBeNull()
+      expect(Object.keys(opts).indexOf('gas')).toBeGreaterThan(-1)
+      expect(Object.keys(opts).indexOf('from')).toBeGreaterThan(-1)
+      expect(Object.keys(opts).indexOf('gasPrice')).toBeGreaterThan(-1)
+      expect(opts['gas']).toBeGreaterThanOrEqual(101191)
```

Comment:
hmm. this is coupled to the estimation in the ABI, but as a magic number here.

if you wanna have the real estimate, then pull it out first with `voting.getGas('methodName')` so that if the ABI changes in the future it won't fail -- and it explains where that number comes from

Reply:
updated the gas expectations